lib/ronad/maybe.rb

```diff
@@ -2,6 +2,12 @@
 
 module Ronad
   class Maybe < Monad
+    # Creates a single Maybe from multiple values, shorting on the first
+    # non-nil value.
+    def self.from_multiple_values(*vals)
+      new vals.detect { |v| Monad.new(v).value != nil }
```

Comment:
Parenthesize the param `vals.detect { |v| Monad.new(v).value != nil }` to make sure that the block will be associated with the `vals.detect` method call.